### PR TITLE
[#218] beta spa

### PR DESCRIPTION
**Pull request by dodoels**
[#218]

- [x] Beta Sign up: Landing Page
- [x] Beta Sign up: Chat and Collect Info
- [x] Beta Sign up: Intro
- [ ] integration to beta back-end
- [x] responsive

image
image
image
image
image

RESPONSIVE VIEW:
image
image
image

### CODE REVIEW

File: src/beta_client/src/components/Beta.vue [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                        <el-col :md="24">
+                            <div id="button-group">
+                                <el-button type="warning" v-on:click="chat = !chat">SIGN UP TO BETA</el-button>
+                            </div>
+                        </el-col>
+                    </el-row>
+                </div>
+            </transition>
+            <div id="beta-page-footer">
+                <a href="http://www.cyberjustice.ca/" target="_blank">© 2017 Cyberjustice Laboratory</a>
```
Comment (lancelafontaine) [inline src/beta_client/src/components/Beta.vue]: I really don't think the `© 2017 Cyberjustice Laboratory` is necessary. It is clear from the `About Us` and Cyberjustice Laboratory owns this project. An alternative would be to make it less subtle (don't like the black text on white background, maybe grey text instead).

File: src/beta_client/src/components/Beta.vue [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                <p>Hosted at the Faculty of Law of the Université de Montréal, its multidisciplinary international research team extends to 45 researchers from 23 institutions in Europe, the Americas, China and Australia, whose investigative goals tend towards improving the allocation and administration of judicial services so as to make optimal use of advances in technology.</p>
+                <p>The Cyberjustice Laboratory is designing an artificial intelligence tool capable of assessing the validity of a litigant’s claim and proposing possible avenues for resolution.</p>
+                <p>The solution takes the shape of a dynamic conversational agent designed to predict the likely outcomes of a variety of landlord/tenant disputes, based on personalized interactions with the disputant and analysis of relevant case law using machine learning algorithms.</p>
+                <p>The project tests the boundaries of machine learning and natural language processing applied to the complex domain of law, and attempts to validate the utility of such a tool for stakeholders.</p>
+            </div>
+            <h2>Engineer Team</h2>
```
Comment (lancelafontaine) [inline src/beta_client/src/components/Beta.vue]: I really think we should remove this. We don't have much to write about, and I don't think anyone will care.

File: src/beta_client/src/components/Chat.vue [hunk 3]
```diff
@@ -4,107 +4,84 @@
...
-      counter: 0,
-      currentMap: [],
-      connectionError: false,
-      regularConversation: [
+      conversation: [
         {
-          user: [{text:'Okay', val: 0}],
-          zeus:'Hello there stranger. My name is Zeus and I\'m here to assist with tenant/landlord problems within Canada. I\'ll be going into beta soon.'
+          user: [{text:'Okay', val: 1}],
+          zeus:'Hello there stranger! My name is Zeus and I\'m here to assist with tenant/landlord problems within Quebec. I\'ll be going into beta soon. Would you like to answer some questions to help me understand better in this field?'
```
Comment (lancelafontaine) [inline src/beta_client/src/components/Chat.vue]: > Hello there stranger! My name is Zeus and I'm here to assist you with your Québec tenant/landlord issues. I'll be going into beta soon, and would like to get more context about common problems. If you don't mind telling me about any tenant/landlord issues you have, we'll try to get back to you as soon as possible with some useful information.

File: src/beta_client/src/components/Chat.vue [hunk 4]
```diff
@@ -4,107 +4,84 @@
...
-          user: [{text:'Sure', val: 0}, {text:'No Question', val: 1}],
-          zeus:'Would you like to ask me a question? I\'ll email you the answer as soon as I can!'},
-        {
-          user: [{text:'Send', val: 3}],
-          zeus:'Great! What is your question?'},
-        {
-          user: [{text:'Sure', val: 4}, {text:'No Thanks', val: 2}],
-          zeus:'I\'ll keep your question in mind. Can I have your email? I\'ll send you the response ASAP!'},
+          user: [{text:'Sure', val: 2}, {text:'I have no question', val: 3}],
+          zeus:'Would you like to tell me a question that you want to have answers the most about tenant/landlord issue?'
```
Comment (lancelafontaine) [inline src/beta_client/src/components/Chat.vue]: I don't understand this question.

File: src/beta_client/src/components/Chat.vue [hunk 5]
```diff
@@ -4,107 +4,84 @@
...
-      ],
-      noQuestionConversation: [
+          user: [{text:'Confirm', val: 3, type: 'question'}],
+          zeus:'Great! What is your question?'
+        },
         {
-          user: [{text:'Sure', val: 0}, {text:'No', val: 2}],
-          zeus:'Would you like to give us your email to know when I have my open beta?'},
+          user: [{text:'Confirm', val: 4, type: 'email'}],
+          zeus:'Can I have your email?'
```
Comment (lancelafontaine) [inline src/beta_client/src/components/Chat.vue]: > Mind leaving your email so that we can let you know once our beta is live?

**Review by lancelafontaine — APPROVED** [review]

**Comment by arekmano** [comment]
We have to have the fewest possible barriers between getting the user's data. IMO

1. We shouldn't have the landing page. It's just a barrier to getting user data
2. We should save every piece of information that users input. If a user enters a question, presses enter , gets bored and leaves the page, we should have his question.
3. We don't need a back button. Unnecessary.

File: src/beta_client/src/components/Chat.vue (hunk 5 above)
Comment (Vynny) [reply src/beta_client/src/components/Chat.vue]: I agree this is a bit too forward. We need to offer the user some value like "If you leave your email, I'll let you know when I go live. I can't wait to help you with any problems you might have!"

**Comment by arekmano** [comment]
Additional UI tweaks that would improve page IMO:

1. Add transparency to the chat container (#beta-page-sign-up) [rgba(74, 74, 74, 0.8)]
2. Reduce chat bubble border radius ~15 px?
3. Add border radius to chat container ~ 10px
4. Have bigger spacing between zeus and input box
5. Darken the chat container border

**Comment by arekmano** [comment]
Used browser-based HTML editing to modify it a little
image

**Comment by dodoels** [comment]
@arekmano i understand that we want to collect the info as efficient as possible, but we have to consider the fact that the first impression of the app is being friendly to the users as well. A landing page is a basic presentation to the new users. Same flow for BotlerAI and other competitors.

**Comment by dodoels** [comment]
for the minor UI tweaks I will leave them to later after collecting feedbacks from everyone. The branding design uses more straight lines and rectangle. I am trying to avoid radius over-use in case of design inconsistency.

File: src/beta_client/src/components/Chat.vue (hunk 4 above)
Comment (dodoels) [reply src/beta_client/src/components/Chat.vue]: Me neither -_ -

**Comment by arekmano** [comment]
In this case, we should eliminate radius in its entirety. Something like this:

image

**Comment by dodoels** [comment]
fix the code and UI based on the feedbacks. I am planning to merge this soon. Few more things to do in next PR
1. integration to BE
2. validation of the answers 
3. improve the questions if needed
4. adjust the send request function based on BE

**Review by arekmano — APPROVED** [review]

**Review by Vynny — APPROVED** [review]

**Pull request merged**